### PR TITLE
Add CONTRIBUTING.md.

### LINKED ISSUE #9687: Add a CONTRIBUTING.md file
Recent wandering into this issue tracker seem to indicate that there's a disconnection between people reporting issues and people fixing them: a lot of the open issues here are actually already fixed.

Even though I'm not sure people actually read them, I suggest adding CONTRIBUTING.md file to entice fixers to have a quick lookup on the issue tracker to see if their fix is not killing a few issues.

This could also be the opportunity to define a minimal format for people reporting issues to minimize the number of message that are needed to be able to issue a diagnostic. 

In particular it would be good to always ask for an `opam config report`. Though usually most of the info there can be inferred by looking at the user file paths (but that may change with opam2 local repos), it's useful to spot usage of internal solver problems and outdated report via the `last-update` field (which would benefit of being a git repository hash rather than a date see https://github.com/ocaml/opam/issues/2982).

### PR DESCRIPTION
**Pull request by dbuenzli**
Closes #9687.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+To report package installation issues:
+
+1. First try to search the issue tracker to see if someone already
+   reported the issue. If that is the case mention on this issue that
+   you are also affected and, if available, add more diagnostic details.
+2. If you create a new issue make sure the issue title mentions the
+   package name and the affected version.
+2. Try to give as much information as possible about your setup. In
+   particular, on installation failures, add the the exact opam
+   install invocation and the resulting tool output to the issue body.
```
Comment (samoht) [inline CONTRIBUTING.md]: even if it the tool responsibility should do it, I think it is worth mentioning `opam config report` here.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+
+Adding new packages
+-------------------
+
+Information about creating new packages and adding them to repository
+is available in
+[opam's manual](https://opam.ocaml.org/doc/Packaging.html).
+
+
```
Comment (samoht) [inline CONTRIBUTING.md]: Can you remove the extra spaces at the end of the file?

**Comment by dbuenzli** [comment]
@samoht's comments were addressed.

**Comment by samoht** [comment]
Thank you!

**Pull request merged**